src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -4,27 +4,55 @@
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
 import me.karakelley.http.Status;
+import me.karakelley.http.utility.HtmlGenerator;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class StaticFilesController implements Controller {
-  private final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+
+  public final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+  private final HtmlGenerator htmlGenerator;
+  private final PublicDirectory publicDirectory;
+
+  public StaticFilesController(PublicDirectory publicDirectory, HtmlGenerator htmlGenerator) {
+    this.htmlGenerator = htmlGenerator;
+    this.publicDirectory = publicDirectory;
+  }
 
   public Response respond(Request request) {
+    String requestedResource = request.getPath();
+
     Response response = new Response();
     if (request.validRequestLine() && availableActions.contains(request.getMethod())) {
-      return displayFilesResponse(response);
+      return serveDirectoryResponse(response, requestedResource);
     } else {
       return new InvalidRequestController().respond(request);
     }
   }
 
-  private Response displayFilesResponse(Response response) {
+  private Response serveDirectoryResponse(Response response, String requestedResource) {
     response.setStatus(Status.OK);
-    response.setBody(String.join("\n", PublicDirectory.getFiles()));
-    response.setHeaders("Content-Type", "text/plain");
+    response.setBody(String.join("", getFilesAndDirectoryLinks(requestedResource)));
```

Comment:
Why join with empty string?

Reply:
It's an list of html elements -  I initially was joining on a newline but the browser renders the html and newlines at that point didn't seem necessary.

Reply:
Ohhhh. This is joining the elements of a list with an empty string. I thought you were just merging two `String`s...one of which was `""`. That makes much more sense.